Add 201 parametric crypto (JCA) misuses

akwick opened the pull request:
All misuses are listed in the *dataset.yml*-file (JCA-Param-All). 

Removed Vidyas dataset through commit b7e53c6.

Commit:
added new datasets

Commit:
changed characteristics to violations

Commit:
update dataset cloudstack

Commit:
updated dataset wicket

Commit:
build commands

Commit:
Findbugs

Commit:
added Findbugs

Commit:
adding Findbugs

Commit:
adding Findbugs

Commit:
config files

anam-dodhy committed:
- added new data sets

anam-dodhy committed:
- deleted data sets: kafka-eagle (compilation error), dragonite-java (unable to build with gradle)
- added new data sets

anam-dodhy committed:
- added new datasets

anam-dodhy committed:
- new data sets created

anam-dodhy committed:
- added 2 new data sets out of which dragonite-java is currently throwing an exception

anam-dodhy committed:
- violation types updated

anam-dodhy committed:
- added correct-usages for selected projects

anam-dodhy committed:
- corrected paths in version.yml and in individual misuses
- correted function definitions in each misuse

anam-dodhy committed:
- updated violation types in jeesuite-libs dataset

anam-dodhy committed:
- added a new data sets for dbeaver-core
- added missing misuse in nettygameserver dataset
- removed unneccesary correct-usages from selected data sets. Now correct-usages only have file with correct usage of the respective misuse.

anam-dodhy committed:
- updated Game-server data set with a missing misuse
- added data sets for gradle projects

anam-dodhy committed:
- removed non gradle projects from the "gradle data" folder
- updated paths in dragonite-java data set

anam-dodhy committed:
- adding google data search metadata file

anam-dodhy committed:
Update metadata.html

anam-dodhy committed:
Update metadata.html

anam-dodhy committed:
Update metadata.html

anam-dodhy committed:
Update metadata.html

anam-dodhy committed:
Update metadata.html

anam-dodhy committed:
- updated google data set metadata

anam-dodhy committed:
Merge branch 'master' of https://github.com/anam-dodhy/MUBench

anam-dodhy committed:
- updated meta data file for google datasets search

anam-dodhy committed:
- added google verification

anam-dodhy committed:
Update README.md

anam-dodhy committed:
- added google verification file

anam-dodhy committed:
Merge branch 'master' of https://github.com/anam-dodhy/MUBench

anam-dodhy committed:
- removed google verification file from root folder
- edited README file to add google verification

anam-dodhy committed:
- edited README file

anam-dodhy committed:
- edited README

anam-dodhy committed:
- dragonite-java gradle data set added

anam-dodhy committed:
- reverted changed to README file

anam-dodhy committed:
- added all data sets for all gradle projects in "gradle data" directory

anam-dodhy committed:
- correct-usage updated for smart dataset

anam-dodhy committed:
- updated violation type for ha-bridge project

anam-dodhy committed:
- configuration file with Bug Patterns to identify parametric cryptographic API-misuses from FindSecBugs Bug patterns list

akwick committed:
Merge branch 'master' of https://github.com/anam-dodhy/MUBench into thesis-2018-dodhy

# Conflicts:
#	data/argouml/misuses/2/misuse.yml
#	data/argouml/misuses/3/misuse.yml
#	data/argouml/misuses/4/misuse.yml
#	data/argouml/misuses/tikanga11-1/correct-usages/HasNext.java
#	data/chensun/misuses/jadet-1a/misuse.yml
#	data/datasets.yml
#	data/synthetic/misuses/hasnext/patterns/HasNext.java
#	data/synthetic_survey/misuses/hasnext/correct-usages/mubench/examples/survey/HasNext.java
#	detectors/Findbugs/configs/core-all-api-misuses.xml

akwick committed:
Remove data which is no collected during thesis

Having only 35 projects is correct as not all of the projects which are buildable includes insecure usages.
In the folder gradle-data the data for the gradle-projects is included.

akwick committed:
Move gradle projects to main project structure

akwick committed:
Add statistics file for all 200 misuses

Proj;API;violation

akwick committed:
Remove dublicated dragonite-java project

akwick committed:
Add schema.org dataset metadata

First try: add some basic schema.org dataset information to the README for our dataset.
Remove old data for Google Dataset search.

Current status:
<https://productforums.google.com/forum/#!topic/webmasters/nPq4BW6iPIA>
[x] Add metadata in schema.org to the README
[x] Verify that markup shows up in the Structured Data Testing Tool
[x] As I don't have multiple pages, I don't need to create a sitemap.

akwick committed:
Add schema.org dataset metadata

First try: add some basic schema.org dataset information to the README for our dataset.

akwick committed:
Merge branch 'thesis-2018-dodhy' of https://github.com/akwick/MUBench into thesis-2018-dodhy

akwick committed:
Add violation category

akwick committed:
Update datasets

akwick committed:
Merge remote-tracking branch 'stg/master' into thesis-2018-dodhy

akwick committed:
Merge branch 'master' of https://github.com/stg-tud/MUBench

Signed-off-by: Anna-Katharina Wickert <wickert@st.informatik.tu-darmstadt.de>

# Conflicts:
#	data/README.md
#	data/argouml/misuses/2/misuse.yml
#	data/argouml/misuses/3/misuse.yml
#	data/argouml/misuses/4/misuse.yml
#	data/argouml/misuses/tikanga11-1/correct-usages/HasNext.java
#	data/chensun/misuses/jadet-1a/misuse.yml
#	data/datasets.yml
#	data/synthetic/misuses/hasnext/patterns/HasNext.java
#	data/synthetic_survey/misuses/hasnext/correct-usages/mubench/examples/survey/HasNext.java
#	data/testng/versions/677302c/version.yml
#	detectors/Findbugs/configs/core-all-api-misuses.xml

akwick committed:
Delete data-sets-exp1 folder

akwick committed:
Merge branch 'master' into thesis-2018-anam-dodhy

Signed-off-by: Anna-Katharina Wickert <wickert@st.informatik.tu-darmstadt.de>

# Conflicts:
#	data/argouml/misuses/tikanga11-1/correct-usages/HasNext.java
#	data/chensun/misuses/jadet-1a/misuse.yml
#	data/datasets.yml
#	data/synthetic_jca/misuses/aes-ecb-mode/misuse.yml
#	data/testng/versions/677302c/version.yml

akwick committed:
Add misuses to version.yml

Add missing misuses to the version.yml file.

akwick committed:
Update Readme

akwick committed:
Remove vidyas misuses

Signed-off-by: Anna-Katharina Wickert <wickert@st.informatik.tu-darmstadt.de>

akwick committed:
Remove vidyas misuses

akwick committed:
Merge branch 'thesis-2018-anam-dodhy' of https://github.com/akwick/MUBench into thesis-2018-anam-dodhy

salsolatragus commented on data/public-cms/versions/f2c72f0/version.yml:
```diff
@@ -0,0 +1,17 @@
+build:
+  classes: publiccms-parent/publiccms-common/target/classes
+  commands:
+  - cd publiccms-parent
```
I think this has no effect, since the commands are executed each in a fresh shell. To `cd` before another command specify one command, such as `cd foo & other-command`.

akwick replied:
Thanks for this information. I have fixed the corresponding yml-file in commit  1a58711 and verified the results with:
````
mubench> pipeline compile --only public-cms
[INFO   ] Starting benchmark...
[INFO   ] All requirements satisfied. You're good to go.
[INFO   ] Compiling project 'public-cms' version f2c72f0...
````

salsolatragus commented on data/testng/versions/677302c/version.yml:
```diff
@@ -9,4 +9,5 @@ misuses:
 - 'dmmc-4'
 - 'grouminer-4'
 - 'grouminer-17'
+- 'mudetect-15'
```
Where's this coming from?

akwick replied:
That's actually a good question. I am sorry. I reset the file to the commit 76edd65d831070b28d5f4b75e09dcf03961ee771 in commit c042a9d80c07808ee9e9f4477398b7c967866613.

salsolatragus commented on data/wordpressa/santuario-xml-security1/misuses/SV_CVE_13_2172/misuse.yml:
```diff
@@ -0,0 +1,20 @@
+api:
```
This file is incomplete.

akwick replied:
This is fixed with commit 5055704a9dac770e31a7e519dff05652aead61d9.

salsolatragus commented on data/wordpressa/santuario-xml-security1/project.yml:
```diff
@@ -0,0 +1,5 @@
+repository:
+  type: git 
+  url: https://github.com/apache/qpid-broker-j.git
+Url: https://github.com/apache/qpid-broker-j.git.   #https://qpid.apache.org
```
Why the dot after the URL and the comment?

akwick replied:
This is fixed with commit 5055704.

salsolatragus commented on data/abixen-platform/misuses/1/misuse.yml:
```diff
@@ -0,0 +1,15 @@
+  file: com/abixen/platform/service/businessintelligence/multivisualisation/domain/model/util/AES128Encoder.java
+  method: "decryptPassword(String)"
+  line: 66
+internal:
```
Please specify `internal: false` for these misuses, as the JCE APIs are not part of any of the projects you added.

akwick replied:
Thanks for pointing this out. I have resolved this in commit 347ce9cb72fe80b0b5a7b7eb68f6c9b9c0f18204.

salsolatragus commented on data/abixen-platform/misuses/1/misuse.yml:
```diff
@@ -0,0 +1,15 @@
+  method: "decryptPassword(String)"
+  line: 66
+internal:
+pattern:
```
This key is obsolete and can be removed.

akwick replied:
Just to clarify: You mean the key `pattern`?

salsolatragus commented on data/abixen-platform/misuses/1/misuse.yml:
```diff
@@ -0,0 +1,15 @@
+violations:
+- insecure/condition/randomization
+description: >
+  First parameter while initializing the IvParameterSpec object was not properly randomized.
```
"not properly randomized" is a fairly vague description of the problem. Can you provide more details, e.g., by explaining how it should be done/could be fixed?

Please us backticks to properly format `type names`.

salsolatragus commented on data/abixen-platform/misuses/2/misuse.yml:
```diff
@@ -0,0 +1,15 @@
+violations:
+- insecure/condition/key
+description: >
+  Second parameter while initializing the Cipher object was not properly generatedKey.
```
"not properly generatedKey" is not a proper sentence, please fix.

salsolatragus commented on data/abixen-platform/misuses/3/misuse.yml:
```diff
@@ -0,0 +1,15 @@
+violations:
+- insecure/condition/transformation
+description: >
+  First parameter in Cipher.getInstance(String) is with value "AES/CBC/PKCS5PADDING" which should be any of AES/CBC/{Empty String, PKCS7Padding, PKCS5Padding, ISO10126Padding}.
```
Is the problem here the incorrect casing of the padding? If so, please state explicitly.

salsolatragus commented on data/abixen-platform/misuses/4/misuse.yml:
```diff
@@ -0,0 +1,15 @@
+violations:
+- insecure/condition/IV
+description: >
+  Third parameter while initializing the Cipher object was not properly preparedIV.
```
"not properly preparedIV" is not a proper sentence, please fix.

salsolatragus commented on data/biglybt-core/misuses/2/misuse.yml:
```diff
@@ -0,0 +1,15 @@
+violations:
+- insecure/condition/randomization
+description: >
+  First parameter while creating the SecretKeySpec objecct i.e. new SecretKeySpec(byte[], String) was not properly randomized
```
"objecct"

salsolatragus commented on data/datasets.yml:
```diff
@@ -788,3 +745,455 @@ JCA-All: # All JCA misuses.
   - tab-apps.62.1
+
+
+JCA-Param-All: # All parametric cryptographic misuses identified in the set.
```
Please mention in a comment that not all of these are compilable.

akwick replied:
I have clarified this in commit 43f5c93ab1a8ec982197dc2f9846d815cb987b07.

salsolatragus reviewed (changes requested):
Thank you very much for preparing this Pull Request. The number of misuse examples you gathered is impressive and a huge contribution to the research community. I'm happy to accept this PR into MUBench, after cleanup.

I'm currently trying to compile the dataset. I'll report if there's any failures. Afterwards, I'll run a test experiment to ensure that all builds produce correct data for running detectors. For the time being, please find detailed comments inline and some more general comments here:

* Please improve the correct usage examples:
    1) They should be minimal examples of correct usage, corresponding to the respective misuse.
    2) They should not be copies of files from other projects, since this might violate respective licenses.
    3) They should not contain Chinese/Japanese comments, commented out code, ...
* Please remove the `.DS_Store` files.
* Please do not make `detectors/Findbugs/configs/core-all-api-misuses.xml` executable.

## Findbugs

* This PR seems to contain data for a second experiment with Findbugs. Can the two be separated into two PRs?
* Please do not add a detector JAR (`Findbugs.jar`) to the benchmark. Detector jars are downloaded during runtime.
* What's the difference between `security-api-misuses.xml` and `security-api-misuse.xml`?

salsolatragus commented:
```bash
[INFO   ] Compiling project 'abixen-platform' version 99fe499...
[ERROR  ] Exception in CompileVersionTask: Failed to execute 'mvn dependency:build-classpath -DincludeScope=compile -pl :abixen-platform-web-content-service -am clean install':
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check (validate) on project abixen-platform-web-content-service: Failed during checkstyle execution: Unable to find configuration file at location: ../../abixen-platform/checkstyle.xml: Could not find resource '../../abixen-platform/checkstyle.xml'. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :abixen-platform-web-content-service
```

akwick committed:
Remove .DS_Store files for PR

akwick committed:
Remove execute permission for core-all-api-misuses.xml

Updated index of the file (git update-index) to change the permission.

akwick committed:
Remove Findbugs.jar

akwick committed:
Remove Vidya's config for FindBugs

akwick committed:
Checkout latest versions of stg-tud/MUBench

To revert the unrelated changes.

akwick commented on README.md:
```diff
@@ -1,4 +1,4 @@
-<img align="right" width="320" height="320" alt="MUBench Logo" src="./meta/logo.png?raw=true" />
+﻿<img align="right" width="320" height="320" alt="MUBench Logo" src="./meta/logo.png?raw=true" />
```
Solved in commit 94b7b3686bdf68b9b5b8e380778b5b0c0d34599b.

akwick commented on data/README.md:
```diff
@@ -249,4 +249,4 @@ Please create a [new issue](https://github.com/stg-tud/MUBench/issues/new) with
 * Instructions how to checkout and compile the project with the misuse from version control.
 * The exact location of the misuse in the project's source code (file, class, and method).
-* (Optional) A link to an issue report that uncovered the misuse.
+* (Optional) A link to an issue report that uncovered the misuse.
```
Solved in commit 94b7b3686bdf68b9b5b8e380778b5b0c0d34599b.

akwick committed:
Replace /target/classes with $mvn.default.classes

Requires no future changes if the default binary path changes.
More details see pullrequest 427 https://github.com/stg-tud/MUBench/pull/427

akwick commented on data/abixen-platform/versions/99fe499/version.yml:
```diff
@@ -0,0 +1,16 @@
+build:
+ classes: abixen-platform-business-intelligence-service/target/classes
```
Resolved for this file and the other 40 files being compiled with mvn and don't use the placeholder in commit e82b7eee77e351484bf5c230f845f8556e59ca35.

salsolatragus commented:
```bash
[INFO   ] Fetching project 'openzaly' version 91bd03b from git:https://github.com/akaxincom/openzaly#91bd03be...
[ERROR  ] Exception in CheckoutTask: Failed to execute 'git clone https://github.com/akaxincom/openzaly . --quiet -c core.askpass=true': === ERROR ===
remote: Repository not found.
fatal: Authentication failed for 'https://github.com/akaxincom/openzaly/'
```

salsolatragus commented:
```bash
[INFO   ] Compiling project 'public-cms' version f2c72f0...
[ERROR  ] Exception in CompileVersionTask: Failed to execute 'mvn dependency:build-classpath -DincludeScope=compile clean compile':
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/mubench/checkouts/public-cms/f2c72f0/build). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException
```

salsolatragus commented:
```bash
[INFO   ] Compiling project 'symmetric-ds' version c42f0e0...
[ERROR  ] Exception in CompileVersionTask: Failed to execute 'gradle :compileJava --debug':
```

salsolatragus commented:
I have build failures due to #409 for another 4 projects. The remaining projects compile fine. Now for running detectors :)

akwick committed:
Add URL to Issue in Dataset

Referenced the mentioned Issue in the dataset.yml with the respective URL of the issue.

Signed-off-by: Anna-Katharina Wickert <wickert@st.informatik.tu-darmstadt.de>

akwick commented on data/datasets.yml:
```diff
@@ -788,3 +745,455 @@ JCA-All: # All JCA misuses.
+  - zheng.11f2e86.2
+  - zheng.11f2e86.3
+
+JCA-Param-WithoutGradle: # All parametric cryptographic misuses without the projects which did not compile in MUBench due to issue #409
```
This is addressed in commit c596f9d5.

akwick committed:
Add source to misuse.yml

akwick committed:
Add URL too

akwick committed:
Add placeholder for gradle build path

akwick committed:
Remove Vidya's dataset

akwick commented on data/argouml/misuses/2/misuse.yml:
```diff
@@ -0,0 +1,28 @@
+one, even though it
+handles all other
+errors.
+location:
```
Resolved in commit  24dadc0 .

akwick commented on data/argouml/misuses/2/misuse.yml:
```diff
@@ -0,0 +1,28 @@
+api:
+- org.argouml.persistence.ZargoFilePersister
+characteristics:
+- missing/exception_handling
```
Resolved in commit  24dadc0 . 
It's a mistake from my side. This is one of Vidya's misuses.

akwick commented on data/argouml/misuses/3/misuse.yml:
```diff
@@ -0,0 +1,28 @@
+api:
+- org.argouml.persistence.ZargoFilePersister
+characteristics:
+- missing/exception_handling
```
Resolved in commit 24dadc0 .
It's a mistake from my side. This is one of Vidya's misuses.

akwick commented on data/argouml/misuses/4/misuse.yml:
```diff
@@ -0,0 +1,26 @@
+- java.util.Iterator
+- java.util.List
+characteristics:
+- misplaced/call
```
Resolved in commit 24dadc0 .
It's a mistake from my side. This is one of Vidya's misuses.

akwick committed:
Change heading datasets.yml

akwick commented on data/datasets.yml:
```diff
@@ -1,51 +1,8 @@
 ###
-### Datasets that group misuses by the source we identified them from.
+### Datasets that group misuses
```
Fixed in commit b9ba545 .

akwick committed:
Add all datasets from stg master

akwick commented on data/datasets.yml:
```diff
@@ -1,51 +1,8 @@
+### Datasets that group misuses
 ###
 
-BugClassify: # All misuses identified in the bug dataset.
```
Addressed in  commit  7ac8a96.

akwick committed:
Add description to findbugs dataset

akwick commented on data/datasets.yml:
```diff
@@ -788,3 +745,455 @@ JCA-All: # All JCA misuses.
+  - zheng.11f2e86.3
+
+
+MSR19-FindBugs: # Misuses used to evaluate FindBugs
```
Add description for the dataset in commit  7ac8a96.

akwick committed:
Minimize correct examples for misuses

Minimize corrected examples for misuses

Minimize correct example for instagram4j

Add correct usages for jeesuite-libs

Minimized corrected examples

Add correct usage for dragonite-java

Add minimal corrected examples

akwick committed:
Minor changes in misuse.yml-files

Add some clarifications or minor changes for the description while
inspecting the misuse.yml for minimizing the corrected examples.

Minor changes in misuse.ymls

Minor change

Minor changes

akwick commented:
> 
> 
> Thank you very much for preparing this Pull Request. The number of misuse examples you gathered is impressive and a huge contribution to the research community. I'm happy to accept this PR into MUBench, after cleanup.
> 
> I'm currently trying to compile the dataset. I'll report if there's any failures. Afterwards, I'll run a test experiment to ensure that all builds produce correct data for running detectors. For the time being, please find detailed comments inline and some more general comments here:
> 
>     * Please improve the correct usage examples:
>       
>       1. They should be minimal examples of correct usage, corresponding to the respective misuse.
>       2. They should not be copies of files from other projects, since this might violate respective licenses.
>       3. They should not contain Chinese/Japanese comments, commented out code, ...
> 
Addressed in commit 8cbfda3a119b99ccb5dc0c9ecf17e58b3c60ad48.
>     * Please remove the `.DS_Store` files.
> 
>     * Please do not make `detectors/Findbugs/configs/core-all-api-misuses.xml` executable.
> 
> 
> ## Findbugs
> 
>     * This PR seems to contain data for a second experiment with Findbugs. Can the two be separated into two PRs?
> 
>     * Please do not add a detector JAR (`Findbugs.jar`) to the benchmark. Detector jars are downloaded during runtime.
> 
>     * What's the difference between `security-api-misuses.xml` and `security-api-misuse.xml`?

akwick commented on data/jeesuite-libs/misuses/1/correct-usages/DES.java:
```diff
@@ -0,0 +1,112 @@
+package com.jeesuite.common.crypt;
```
This comment is addressed in commit 8cbfda3a119b99ccb5dc0c9ecf17e58b3c60ad48.

akwick committed:
Fix cmd commands for public-cms

Signed-off-by: Anna-Katharina Wickert <wickert@st.informatik.tu-darmstadt.de>

akwick committed:
Reset to commit 76edd65d831070b28d5f4b75e09dcf03961ee771

akwick committed:
Reset to 8d9e1faa565b7fc59ea6f1c6ae82e8a251d32fdc for worpressa

Signed-off-by: Anna-Katharina Wickert <wickert@st.informatik.tu-darmstadt.de>

akwick committed:
Specify internal flag to false in misuse files

All identified misuses are due to the JCA and the JCA isn't part of any
project added. Thus, the flag should be false.

akwick committed:
Add information not all projects compilable

The complete dataset contains some projects which are currently not
compilable.

akwick commented:
> 
> 
> ```shell
> [INFO   ] Compiling project 'abixen-platform' version 99fe499...
> [ERROR  ] Exception in CompileVersionTask: Failed to execute 'mvn dependency:build-classpath -DincludeScope=compile -pl :abixen-platform-web-content-service -am clean install':
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check (validate) on project abixen-platform-web-content-service: Failed during checkstyle execution: Unable to find configuration file at location: ../../abixen-platform/checkstyle.xml: Could not find resource '../../abixen-platform/checkstyle.xml'. -> [Help 1]
> [ERROR]
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR]
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
> [ERROR]
> [ERROR] After correcting the problems, you can resume the build with the command
> [ERROR]   mvn <goals> -rf :abixen-platform-web-content-service
> ```

abixen-platform is already removed from the set of the projects which didn't compile currently.

akwick committed:
Remove openzaly

We have to remove the two misuses of openzaly as this project isn't
available any more on GitHub.

akwick committed:
Remove openzaly

We have to remove the two misuses of the openzaly projects as the project isn't publicy available any more.

Remove openzaly

We have to remove the two misuses of openzaly as this project isn't
available any more on GitHub.

Remove openzaly data

akwick committed:
Merge branch 'thesis-2018-anam-dodhy' of https://github.com/akwick/MUBench into thesis-2018-anam-dodhy

akwick commented:
> 
> 
> ```shell
> [INFO   ] Fetching project 'openzaly' version 91bd03b from git:https://github.com/akaxincom/openzaly#91bd03be...
> [ERROR  ] Exception in CheckoutTask: Failed to execute 'git clone https://github.com/akaxincom/openzaly . --quiet -c core.askpass=true': === ERROR ===
> remote: Repository not found.
> fatal: Authentication failed for 'https://github.com/akaxincom/openzaly/'
> ```

:( I had to remove this project from the dataset in commit db6c7b228708ecffe6408d228c78bf0e3b7d4753.

akwick commented:
> 
> 
> ```shell
> [INFO   ] Compiling project 'public-cms' version f2c72f0...
> [ERROR  ] Exception in CompileVersionTask: Failed to execute 'mvn dependency:build-classpath -DincludeScope=compile clean compile':
> [ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/mubench/checkouts/public-cms/f2c72f0/build). Please verify you invoked Maven from the correct directory. -> [Help 1]
> [ERROR]
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR]
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException
> ```

I can't reproduce this result. :/

````
mubench> pipeline compile --only public-cms
[INFO   ] Starting benchmark...
[INFO   ] All requirements satisfied. You're good to go.
````

akwick committed:
Fix build cmds

akwick commented:
> 
> 
> ```shell
> [INFO   ] Compiling project 'symmetric-ds' version c42f0e0...
> [ERROR  ] Exception in CompileVersionTask: Failed to execute 'gradle :compileJava --debug':
> ```

Fix this issue in commit 2a5971f28aa12a2da44b6f9e4379f2e5d1728157. 

````
mubench> pipeline compile --force-compile --only  symmetric-ds
[INFO   ] Starting benchmark...
[INFO   ] All requirements satisfied. You're good to go.
[INFO   ] Compiling project 'symmetric-ds' version c42f0e0...
````

akwick committed:
Fix ha-bridge minimized examples

mubench> pipeline compile --only ha-bridge --force-compile
[INFO   ] Starting benchmark...
[INFO   ] All requirements satisfied. You're good to go.
[INFO   ] Compiling project 'ha-bridge' version c25f08f...
[INFO   ] Compiling correct usage for misuse 'ha-bridge.c25f08f.2'...
[INFO   ] Compiling correct usage for misuse 'ha-bridge.c25f08f.3'...
[INFO   ] Compiling correct usage for misuse 'ha-bridge.c25f08f.4'...
[INFO   ] Compiling correct usage for misuse 'ha-bridge.c25f08f.5'...
[INFO   ] Compiling correct usage for misuse 'ha-bridge.c25f08f.6'...
[INFO   ] Compiling correct usage for misuse 'ha-bridge.c25f08f.7'...
[INFO   ] Compiling correct usage for misuse 'ha-bridge.c25f08f.8'..

akwick committed:
Fix dragonite-java

akwick committed:
Fix instagram4j

pipeline compile --only instagram4j
[INFO   ] Starting benchmark...
[INFO   ] All requirements satisfied. You're good to go.
[INFO   ] Compiling correct usage for misuse 'instagram4j.ae85b2b.1'...

akwick committed:
Fix jeesuite-libs

mubench> pipeline compile --only jeesuite-libs
[INFO   ] Starting benchmark...
[INFO   ] All requirements satisfied. You're good to go.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Compiling correct usage for misuse 'jeesuite-libs.2a545bd.3'...
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.

akwick committed:
Fix nettygameserver

mubench> pipeline compile --only nettygameserver
[INFO   ] Starting benchmark...
[INFO   ] All requirements satisfied. You're good to go.
[INFO   ] Correct usage already compiled.
[INFO   ] Compiling correct usage for misuse 'nettygameserver.c069be1.2'...
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.

akwick committed:
Fix smart

mubench> pipeline compile --only smart
[INFO   ] Starting benchmark...
[INFO   ] All requirements satisfied. You're good to go.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Compiling correct usage for misuse 'smart.9e018a6.6'...
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled

akwick committed:
Fix correct usages

mubench> pipeline compile --only whatsmars
[INFO   ] Starting benchmark...
[INFO   ] All requirements satisfied. You're good to go.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Compiling correct usage for misuse 'whatsmars.917b029.6'...
[INFO   ] Correct usage already compiled.
[INFO   ] Compiling correct usage for misuse 'whatsmars.917b029.8'...
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.
[INFO   ] Correct usage already compiled.

akwick committed:
Merge branch 'stg-tud:master' into thesis-2018-anam-dodhy
